manifests/forwarder/service.pp

```diff
@@ -4,10 +4,15 @@
 #   sub-classes
 #
 class splunk::forwarder::service {
+  $provider = $facts['os']['family'] ? {
```

Comment:
why is that required? Did you run into any issues?

Reply:
When running de rspec test locally (on an ubuntu 22.04 bundle exec rake spec) the test on windows  fails with the error Provider redhat must have features 'delayed_startable'
I have no idea why the provider redhat was used for the service resource in the rspec test for windows but this seems to fix it.

Reply:
I removed the code and on puppet7 with ruby 2.7 it fails
Error: splunk::forwarder supported operating systems on windows-2012-x86_64 is expected to compile into a catalogue without dependency cycles

Failure/Error: it { is_expected.to compile.with_all_deps }
  error during compilation: Parameter enable failed on Service[SplunkForwarder]: Provider redhat must have features 'delayed_startable' to set 'enable' to 'delayed' (file: /home/runner/work/puppet-splunk/puppet-splunk/spec/fixtures/modules/splunk/manifests/forwarder/service.pp, line: 7)
  
  Do you have an other suggestion to fix this or shall i put the selector back in the code

Reply:
Fixed it by using $facts['service_provider'] as provider and add the fact to the rspec on windows